api-server/src/server/boot/user.js

```diff
@@ -93,21 +93,23 @@ function deleteUserTokenResponse(req, res) {
 }
 
 function createReadSessionUser(app) {
-  const { Donation } = app.models;
+  const { Donation, UserToken } = app.models;
 
   return async function getSessionUser(req, res, next) {
     const queryUser = req.user;
 
-    const userTokenArr = await queryUser.userTokens({
-      userId: queryUser.id
-    });
-
-    const userToken = userTokenArr[0]?.id;
     let encodedUserToken;
 
-    // only encode if a userToken was found
-    if (userToken) {
-      encodedUserToken = encodeUserToken(userToken);
+    if (queryUser && queryUser.id) {
+      // if queryUser is undefined, this returns the first document
```

Comment:
I'm confused by this comment - if `queryUser` is undefined, this query wouldn't execute at all?

Reply:
I added the `if (queryUser...)` cause the db query returns the first doc if you try to use undefined as the `userId`: `where: { userId: queryUser.id }`

Reply:
How about

```js
    const userId = queryUser?.id;
    const userToken = userId
      ? await UserToken.findOne({
          where: { userId }
        })
      : null;

    const encodedUserToken = userToken
      ? encodeUserToken(userToken.id)
      : undefined;
```

I think this code's fairly self-explanatory.

Reply:
✅

Reply:
```javascript
const encodedUserToken =  queryUser?.id ? await UserToken.findOne({where: {userId: queryUser.id}}) ? encodeUserToken(queryUser?.id ? await UserToken.findOne({where: {userId: queryUser.id}}).id : null) : undefined
```

Reply:
Don't make me remove your maintainer access Shaun